### PR TITLE
fix: [PL-29637]: Added template and template studio routes

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/RouteDestinations.tsx [hunk 1]
```diff
@@ -354,6 +355,14 @@ export default (
       <TemplatesPage />
     </RouteWithLayout>
 
+    <RouteWithLayout
```
Comment (abhinavrastogi-harness) [inline src/modules/85-cv/RouteDestinations.tsx:358]: why is this added in `85-cv`?
it should be 45-projects or similar.

File: src/modules/72-triggers/RouteDestinations.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export const TriggersRouteDestinations: React.FC<{
       exact
       licenseRedirectData={licenseRedirectData}
       sidebarProps={ProjectDetailsSideNavProps}
-      path={routes.toTriggersDetailPage({ ...accountPathProps, ...triggerPathProps, ...moduleParams })}
+      path={routes.toTriggersDetailPage({ ...accountPathProps, ...triggerPathProps })}
```
Comment (abhinavrastogi-harness) [inline src/modules/72-triggers/RouteDestinations.tsx:67]: project scope route destination should be in 45-projects only

File: src/modules/10-common/navigation/MainNav/MainNav.tsx [hunk 3]
```diff
@@ -67,11 +67,27 @@ export default function L1Nav(): React.ReactElement {
   })
 
   useEffect(() => {
-    if (!modulesPreferenceData?.orderedModules?.length && NEW_LEFT_NAVBAR_SETTINGS) {
-      const modulesWithLicense = DEFAULT_MODULES_ORDER.filter(m => !!moduleMap[m].hasLicense)
+    if (NEW_LEFT_NAVBAR_SETTINGS) {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/navigation/MainNav/MainNav.tsx:70]: what's this refactor for?

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/10-common/navigation/MainNav/MainNav.tsx (hunk 3 above)
Comment (abhinavrastogi-harness) [reply src/modules/10-common/navigation/MainNav/MainNav.tsx:70]: okay. please add comments in the code to explain the business logic here

File: src/modules/85-cv/RouteDestinations.tsx (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/modules/85-cv/RouteDestinations.tsx:358]: `TemplatesPage` comes from `@templates-library` module. .the route destination should be in that module.. not in CV

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**